### ISSUE TITLE
GLMakie precompile hangs on Windows 10

### ISSUE THREAD

**Issue by Ininterrompue** [MakieOrg/Makie.jl#2941]
After updating to Julia 1.9, the following is shown in the REPL after `]up GLMakie`:

```
Precompiling project...
  Progress [>                                        ]  0/1
  ◒ GLMakie
```

After a bit, CPU usage hovers around 0%. CairoMakie precompiled just fine.

GLMakie precompiles just fine on 1.8.5. 

Attempted reinstalling GLMakie, restarting Julia, rebooting computer, using juliaup. Using a 64-bit system.

**Comment by SimonDanisch** [comment]
I think you should open an issue with Julia for this, I don't think that's anything we can fix in Makie.

**Comment by SimonDanisch** [comment]
Probably related: https://github.com/MakieOrg/GraphMakie.jl/issues/130#issuecomment-1542513437

**Comment by olejorik** [comment]
I have the same issue, starting from julia-1.9 beta2. The workaround I use currently is to `build GLMakie` first. 

I'm on Windows 11 developer, so I don't know whether the problem might be caused by some drivers, because when the precompilation fails, a number of other applications become slow or fail to display their screen correctly until I restart the system.

**Comment by olejorik** [comment]
sure:

Edition	Windows 11 Pro Insider Preview
Version	22H2
Installed on	‎24-‎4-‎2023
OS build	25346.1001
Experience	Windows Feature Experience Pack 1000.25346.1001.0

Processor	Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz   2.70 GHz
Installed RAM	16,0 GB (15,9 GB usable)
System type	64-bit operating system, x64-based processor

**Comment by SimonDanisch** [comment]
Hm, `]build GLMakie` should really not do much... Is that reproducable, or could just be a fluke? 
Anyways, I think this must be more of a windows + julia issue, if it started happening on some specific julia version...
Funnily enough, I'm on windows and have been on all 1.9 versions and never run into this....

**Comment by olejorik** [comment]
Well, just gave it a try by deleting `.julia/compiled/v.1.9` folder and running `] update`. 
![image](https://github.com/MakieOrg/Makie.jl/assets/12269930/afb15ba1-93f6-4dfc-9e45-f1fe151f2753)
Everything worked fine...

**Comment by SimonDanisch** [comment]
I guess that could help indeed when upgrading from a beta to the the released version!

**Comment by Ininterrompue** [comment]
`build GLMakie` does not work for me. I encounter the same hang with CPU usage going to 0%.

**Comment by SimonDanisch** [comment]
Did you try deleting `~/.julia/compiled/v.1.9`?

**Comment by Ininterrompue** [comment]
Yes, I should have clarified that the compiled folder was deleted before `build Makie` was run again.

**Comment by olejorik** [comment]
> `build GLMakie` does not work for me. I encounter the same hang with CPU usage going to 0%.

Well, then it's quite different from what I have experienced. In my case, when the precompilation got stuck, the memory usage by julia process kept growing and the process became "immortal", that is I was not able to kill it even from the terminal.

With the `build` process, it stuck too, but then I was able to terminate it from within julia with Ctrl+C and restart the build again. It always started from ModernGL, and then proceeded with GLMakie (so, in this case, there were always these two packages left).

When trying to solve the issue, I've added `juliaup`, `launchjulia`, together wiht `julia` from `1.9\bin` to the exclusion list for Windows Defender. Maybe this is the reason that I cannot reproduce the issue now. I'll try it later during the coming long weekend and will let you know.
